packages/apify/src/charging.ts

```diff
@@ -131,40 +184,25 @@ export class ChargingManager {
                 );
             }
 
-            const run = await this.apifyClient.run(this.actorRunId).get();
-            if (run === undefined) {
-                throw new Error('Actor run not found');
-            }
-
-            this.pricingModel = run.pricingInfo?.pricingModel;
-
-            // Load per-event pricing information
-            if (run.pricingInfo?.pricingModel === 'PAY_PER_EVENT') {
-                for (const [eventName, eventPricing] of Object.entries(
-                    run.pricingInfo.pricingPerEvent.actorChargeEvents,
-                )) {
-                    this.pricingInfo[eventName] = {
-                        price: eventPricing.eventPriceUsd,
-                        title: eventPricing.eventTitle,
-                    };
+            if (
+                this.chargingState === undefined ||
```

Comment:
how about removing the `load*` methods from the constructor, running them only once in the `init` method (at one call site) and conditionally loading the config with something like

```javascript
async function getPricingInfo() {
   if (
        configuration.get('actorPricingInfo') &&
        configuration.get('chargedEventCounts')
    ) return { pricingInfo: JSON.parse(), ...

    const run = await this.apifyClient.run(this.actorRunId).get();
    ...
}
```

This would allow us to load the pricing info + events at one point in `init` like

```
...
const { pricingInfo ... } = await getPricingInfo();
this.loadPricingInfo( pricingInfo );
...
```

which is IMO cleaner than the current approach, which is conditional based on the current object state